src/utils/messageGeneration/fields.js

```diff
@@ -283,7 +283,7 @@ fields.getPrimitiveSize = function(fieldType, fieldValue) {
   }
   else if (fieldType === 'string') {
     if (fieldValue !== undefined) {
-      fieldSize = fieldValue.length + 4;
+      fieldSize = Buffer.byteLength(fieldValue, 'utf-8') + 4;
```

Comment:
Minor, but should this be `'utf8'` instead of `'utf-8'` ([docs](https://nodejs.org/api/buffer.html#buffer_buffers_and_character_encodings))? It's unclear to me if this is an accepted alternative or if it's falling back to the default value (`'utf8'`) because it doesn't recognize the provided option.